smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/customtraits/RecursiveUnionTrait.kt

```diff
@@ -0,0 +1,12 @@
+package software.amazon.smithy.swift.codegen.customtraits
+
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.Trait
+
+class RecursiveUnionTrait : Trait {
```

Comment:
This trait is added during preprocessing to a union that should be marked `indirect`.